src/user/index.ts

```diff
@@ -217,13 +217,59 @@ export const getUser = async (uuid: string) => {
 			...(await Oauth.getAuthHeaders())
 		}
 	})
-	if(result.status !== 200) {
-		throw new AuthError("Unable to get user");
+	if(result.status === 403) {
+		throw new ApiGetError("You do not have access to view this user. (403 Forbidden)");
+	}
+	else if(result.status !== 200) {
+		throw new ApiGetError("We were unable to get information about this user.");
 	}
 
 	return await result.json() as FullUser;
 }
 
+export const modifyUser = async (
+	uuid: string,
+	firstname?: string, 
+	lastname?: string, 
```

Comment:
There is a big problem with this: Since javascript doesn't support keyword arguments, if you want to f.ex modify `address`, you have to provide `undefined` for all the prior arguments.

Maybe it's better for the values to be provided in a type object instead, that we simply stringify together with the uuid.

That is:

```ts
type ModifyUserKeys = "firstname" | "lastname" | "username" | "email" | "phone" | "guardian_phone" | "address" | "postal_code" | "birthdate" | "gender";
export const modifyUser = async (
    uuid: string,
    values?: Record<ModifyUserKeys, string>) => {
        const response = await fetch(`${getApiServer()}/user/${uuid}`, {
	method: 'PATCH',
	headers: {
		'Content-Type': 'application/json',
		...(await Oauth.getAuthHeaders())
	},
	body: JSON.stringify({
		uuid,
		...values
	})})
} 
```

Also, are all types supposed to be strings?

Reply:
Changed the code for modifyUser to what you suggested!
And I think so, should some of the arguments be something else?